### PR TITLE
PMM-5052 documentation about textfile collector usage in pmm2

### PR DESCRIPTION
**Pull request by adivinho**

### CODE REVIEW

File: docs/how-to/extend-metrics.md [hunk 1]
```diff
@@ -1,4 +1,36 @@
...
+| Resolution | Folder                                                                  |
+|------------|-------------------------------------------------------------------------|
+|  High      | /usr/local/percona/pmm2/collectors/textfile-collector/high-resolution   |
+|  Medium    | /usr/local/percona/pmm2/collectors/textfile-collector/medium-resolution |
+|  Low       | /usr/local/percona/pmm2/collectors/textfile-collector/low-resolution    |
+
+![!image](../_images/node-exporter.textfile-collector.1.png)
+
+Exporter parse all files in that directories matching the glob *.prom using the text format.
+
```
Comment (rnovikovP) [inline docs/how-to/extend-metrics.md:19]:
```suggestion
Files in the folders are expected to be parsed in [Exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/) and metrics stored on PMM Server-side with an additional label related to this Node.

```

Comment (adivinho) [reply docs/how-to/extend-metrics.md:20]: I have added the link and the last sentence.

Screenshot 2021-09-29 at 11 56 42

File: docs/how-to/extend-metrics.md [hunk 2]
```diff
@@ -1,4 +1,37 @@
 # Extend Metrics
 
+Sometimes, though, you’ll have the need for a metric that’s not present in the list of node_exporter metrics out of the box.
```
Comment (PaulJacobs-percona) [inline docs/how-to/extend-metrics.md:3]:
```suggestion
When you need a metric that’s not present in the default list of `node_exporter` metrics you may be able to use the `textfile` collector. 
```

File: docs/how-to/extend-metrics.md [hunk 3]
```diff
@@ -1,4 +1,37 @@
 # Extend Metrics
 
+Sometimes, though, you’ll have the need for a metric that’s not present in the list of node_exporter metrics out of the box. 
+The textfile collector allows exporting of statistics from batch jobs. It can also be used to export static metrics, such as what role a machine has. 
+
 ## Enable the textfile collector
 
+The collector is enabled by default. Next folders are used for different resolutions:
```
Comment (PaulJacobs-percona) [inline docs/how-to/extend-metrics.md:8]:
```suggestion
The collector is enabled by default. The following folders are used for different resolutions:
```

File: docs/how-to/extend-metrics.md [hunk 4]
```diff
@@ -1,4 +1,37 @@
...
+
+| Resolution | Folder                                                                  |
+|------------|-------------------------------------------------------------------------|
+|  High      | /usr/local/percona/pmm2/collectors/textfile-collector/high-resolution   |
+|  Medium    | /usr/local/percona/pmm2/collectors/textfile-collector/medium-resolution |
+|  Low       | /usr/local/percona/pmm2/collectors/textfile-collector/low-resolution    |
+
+![!image](../_images/node-exporter.textfile-collector.1.png)
+
+Exporter parse all files in that directories matching the glob *.prom using a simple text-based [exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format).
```
Comment (PaulJacobs-percona) [inline docs/how-to/extend-metrics.md:18]:
```suggestion
The exporter parses all files in these directories that match the filename wildcard expression `*.prom` using a simple text-based [exposition format](https://prometheus.io/docs/instrumenting/exposition_formats/#text-based-format).
```

File: docs/how-to/extend-metrics.md [hunk 5]
```diff
@@ -1,4 +1,37 @@
...
+Metrics are stored on PMM Server-side with additional labels related to this Node.
```
Comment (PaulJacobs-percona) [inline docs/how-to/extend-metrics.md:19]:
```suggestion
Metrics are stored on the PMM Server-side with additional labels related to this Node.
```

File: docs/how-to/extend-metrics.md [hunk 6]
```diff
@@ -1,4 +1,37 @@
...
+
+## Examples of shell commands for custom metrics
+
+To statically set roles for a machine using labels:
+
+    ```sh
```
Comment (PaulJacobs-percona) [inline docs/how-to/extend-metrics.md:25]:
```suggestion
```sh
```

File: docs/how-to/extend-metrics.md [hunk 7]
```diff
@@ -1,4 +1,37 @@
...
+
+```sh
+    echo 'node_role{role="my_monitored_server_1"} 1' > /usr/local/percona/pmm2/collectors/textfile-collector/low-resolution/node_role.prom
+    ```
+
+To atomically push loggedin users for a cron job:
+
+    ```sh
+    $ cat /etc/cron.d/loggedin_users
+    */1 * * * *     root    /usr/bin/who | /usr/bin/wc -l | sed -ne 's/^/node_loggedin_users /p' > /usr/local/percona/pmm2/collectors/textfile-collector/high-resolution/node_users.prom
```
Comment (PaulJacobs-percona) [inline docs/how-to/extend-metrics.md]: This command doesn't look right to me. The output from `cat /etc/cron.d/loggedin_users` is what? Then the part beginning `/usr/bin/who ...` comes where?

Comment (PaulJacobs-percona) [reply docs/how-to/extend-metrics.md]: When fixed, this code block doesn't need to be indented.

Comment (adivinho) [reply docs/how-to/extend-metrics.md]: It's an example of a cron job. I can remove the "cat" part but I think people would be more confused. 

Let's add the next screenshot.
Screenshot 2021-09-29 at 12 23 06

**Review by rnovikovP — APPROVED** [review]

File: docs/how-to/extend-metrics.md (hunk 7 above)
Comment (PaulJacobs-percona) [reply docs/how-to/extend-metrics.md]: I see, the whole pipeline is one big cron job...

File: docs/how-to/extend-metrics.md [hunk 8]
```diff
@@ -1,4 +1,37 @@
...
+
+## Examples of shell commands for custom metrics
+
+To statically set roles for a machine using labels:
+
+```sh
+echo 'node_role{role="my_monitored_server_1"} 1' > /usr/local/percona/pmm2/collectors/textfile-collector/low-resolution/node_role.prom
+```
+
+To atomically push loggedin users for a cron job:
```
Comment (PaulJacobs-percona) [inline docs/how-to/extend-metrics.md:29]:
```suggestion
Here's an example of a cron job that atomically pushes logged-in users:
```

**Review by PaulJacobs-percona — APPROVED** [review]

**Pull request merged**